app/src/main/java/com/example/capstoneproject/ui/main/CurrencyDetailsFragment.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.capstoneproject.ui.main
+
+import android.os.Bundle
+import android.util.Log
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.navArgs
+import com.bumptech.glide.Glide
+import com.example.capstoneproject.databinding.FragmentCurrencyDetailsBinding
+
+class CurrencyDetailsFragment : Fragment() {
+    private var bookName: String? = null
+
+    private var _binding: FragmentCurrencyDetailsBinding? = null
+    private val binding get() = _binding!!
+
+    private val mainViewModel: MainViewModel by viewModels()
+
+    private val args: CurrencyDetailsFragmentArgs by navArgs()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        bookName = args.bookName
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentCurrencyDetailsBinding.inflate(inflater, container, false)
+        binding.bookNameText.text = bookName
+
+        bookName?.let { mainViewModel.getBookInfo(it) }
+        val imageUrl: String = "https://cryptoicon-api.vercel.app/api/icon/" +
+                bookName?.substringBefore("_", "")
+        Glide.with(binding.currencyImageView.context).load(imageUrl).fitCenter()
+            .into(binding.currencyImageView)
+
+        mainViewModel.tickerInfo.observe(viewLifecycleOwner) { ticker ->
+            Log.e("BITSO", "Max price" + ticker.high_price)
+            binding.highestTextView.text = ticker.high_price
+            binding.lowestTextView.text = ticker.low_price
+            binding.lastTextView.text = ticker.last_price
+        }
+        // Inflate the layout for this fragment
+        return binding.root
+    }
+}
```

Comment:
could you please override the onDestroyView method and set to null the _binding attribute? :D

Reply:
Done, thanks!